src/Compilers/Core/Portable/CommandLine/CommonCommandLineArguments.cs

```diff
@@ -287,7 +287,7 @@ public CompilationOptions CompilationOptions
             string tempDirectory)
         {
             bool fallback =
-                !CoreClrShim.IsRunningOnCoreClr ||
+                !(CoreClrShim.IsRunningOnCoreClr || Type.GetType("Mono.Runtime") != null) ||
```

Comment:
this should eventually be refactored into a helper since it's used in a couple places in the codebase: https://github.com/dotnet/roslyn/search?q=%22Mono.Runtime%22&type=Code

Reply:
Need to wrap this into a helper for this change I think. The `Type.GetType` method can throw exceptions in corner cases. Can't let those take down the compiler here.